sqflite_common/lib/sqlite_api.dart

```diff
@@ -31,64 +34,80 @@ abstract class DatabaseFactory {
-  /// @param table The table names to compile the query against.
-  /// @param columns A list of which columns to return. Passing null will
+  /// [table]: The table names to compile the query against.
+  ///
+  /// [columns]: A list of which columns to return. Passing null will
   ///            return all columns, which is discouraged to prevent reading
   ///            data from storage that isn't going to be used.
-  /// @param where A filter declaring which rows to return, formatted as an SQL
+  ///
+  /// [where]: A filter declaring which rows to return, formatted as an SQL
   ///            WHERE clause (excluding the WHERE itself). Passing null will
   ///            return all rows for the given URL.
-  /// @param groupBy A filter declaring how to group rows, formatted as an SQL
+  ///
+  /// [groupBy]: A filter declaring how to group rows, formatted as an SQL
   ///            GROUP BY clause (excluding the GROUP BY itself). Passing null
   ///            will cause the rows to not be grouped.
-  /// @param having A filter declare which row groups to include in the cursor,
+  ///
+  /// [having]: A filter declare which row groups to include in the cursor,
   ///            if row grouping is being used, formatted as an SQL HAVING
   ///            clause (excluding the HAVING itself). Passing null will cause
   ///            all row groups to be included, and is required when row
   ///            grouping is not being used.
-  /// @param orderBy How to order the rows, formatted as an SQL ORDER BY clause
+  ///
+  /// [orderBy]: How to order the rows, formatted as an SQL ORDER BY clause
   ///            (excluding the ORDER BY itself). Passing null will use the
   ///            default sort order, which may be unordered.
-  /// @param limit Limits the number of rows returned by the query,
-  /// @param offset starting index,
-
-  /// @return the items found
-  Future<List<Map<String, dynamic>>> query(String table,
-      {bool distinct,
-      List<String> columns,
-      String where,
-      List<dynamic> whereArgs,
-      String groupBy,
-      String having,
-      String orderBy,
-      int limit,
-      int offset});
+  ///
+  /// [limit]: Limits the number of rows returned by the query,
+  ///
+  /// [offset]: starting index.
+  ///
+  /// returns the items found.
+  Future<List<Map<String, Object?>>> query(String table,
+      {bool? distinct,
+      List<String>? columns,
+      String? where,
+      List<Object>? whereArgs,
+      String? groupBy,
+      String? having,
+      String? orderBy,
+      int? limit,
+      int? offset});
 
-  /// Execute a raw SQL SELECT query
+  /// Execute a raw SQL SELECT query.
   ///
-  /// Returns a list of rows that were found
-  Future<List<Map<String, dynamic>>> rawQuery(String sql,
-      [List<dynamic> arguments]);
+  /// Returns a list of rows that were found.
+  Future<List<Map<String, Object?>>> rawQuery(String sql,
+      [List<Object>? arguments]);
```

Comment:
Using `List<Object>?` disallows querying for `null` values which was possible before (also applies to other functions). Is this planned?

Reply:
Thanks for the review. I don't think you can bind null value in SQLite. You must explicitly use `IS NULL` and not `= ?` in this case. Well I could be wrong but do you have an example of a working rawQuery binding null? sqlite3 has the same parameter types (`List<Object>` and not `List<Object?>`)

Reply:
> Well I could be wrong but do you have an example of a working rawQuery binding null?

Nope. Binding `null` definitely results in no good and using Dart's newly-gained null safety will allow us to avoid these mistakes. It's still a change in the public API of the library and I was running into some questions while migrating from the not null safe version to this one.

Reply:
@vitusortner after talking with Simon Binder (see https://github.com/tekartik/sqflite/issues/529) it sounds like I will switch to `List<Object?>`. Sorry for the confusion.

Reply:
No worries. Thanks for the update 🙌